go_stash/groups_test.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2021 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gostash
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+)
+
+func TestGetGroup(t *testing.T) {
+	tests := []struct {
+		name      string
+		groupName string
+		output    string
+	}{
+		{
+			name:      "test group does not exist",
+			groupName: "admin",
+		},
+		{
+			name:      "test a group",
+			groupName: "dev-1",
+		},
+	}
+
+	validNames := []string{"dev-1"}
+
+	mux, server, client := setup(t)
+	defer teardown(server)
```

Comment:
You can simplify this by adding a [`t.Cleanup()`](https://pkg.go.dev/testing#T.Cleanup) in the `setup` function

If you modified `func setup` like this, you wouldn't need to teardown...it would be done when the test completed:

```go
// setup sets up a test HTTP server along with a Client configured to talk to that test server.
// Tests should register handlers on mux which provide mock responses for the API method being tested.
func setup(t *testing.T) (*http.ServeMux, *httptest.Server, *Client) {
    // I'm leaving out the rest of the code for clarity.
    t.Cleanup(func() {
      server.Close()
    })
    return mux, server, client
}
```

I should also point out that in `func setup` `berearHeader` is misspelled, it should be `bearerHeader` and I think that...

```go
if err != nil {
    server.Close()
    t.Errorf("unexpected error while declaring a client: %v", err)
}
```

I think that should be `t.Fatalf` because the `*Client` is unusable, which would mean you didn't need to `server.Close()` because the `t.Cleanup()` would take care of closing the server.